src/test/java/api/support/APITests.java

```diff
@@ -251,105 +257,47 @@
   protected final FeeFineAccountFixture feeFineAccountFixture = new FeeFineAccountFixture();
 
   protected APITests() {
-    this(true);
+    this(true, false);
   }
 
-  protected APITests(boolean initialiseCirculationRules) {
+  protected APITests(boolean initialiseCirculationRules, boolean enableLoanHistory) {
     this.initialiseCirculationRules = initialiseCirculationRules;
+    setLoanHistoryEnabled(enableLoanHistory);
   }
 
   @BeforeClass
   public static void beforeAll() throws InterruptedException, ExecutionException,
     TimeoutException {
 
-    deployVerticles();
+    if (okapiAlreadyDeployed) {
+      return;
+    }
 
-    //Delete everything first just in case
-    deleteAllRecords();
+    deployVerticles();
+    Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+      try {
+        undeployVerticles();
+      } catch (Exception ex) {
+        Assert.fail("Failed to undeploy verticle: " + ex);
+      }
+    }));
+    okapiAlreadyDeployed = true;
   }
 
   @Before
   public void beforeEach() throws InterruptedException {
-    requestsClient.deleteAll();
-    loansClient.deleteAll();
-
-    itemsClient.deleteAll();
-    holdingsClient.deleteAll();
-    instancesClient.deleteAll();
-    configClient.deleteAll();
-    accountsClient.deleteAll();
-    feeFinesClient.deleteAll();
-    feeFineOwnersClient.deleteAll();
-    feeFineActionsClient.deleteAll();
-
-    //TODO: Only cleans up reference records, move items, holdings records
-    // and instances into here too
-    itemsFixture.cleanUp();
-
-    usersClient.deleteAllIndividually();
-
-    checkInOperationClient.deleteAll();
-
```

Comment:
Why are records no longer deleted before each test?

Reply:
It is replaced by removing data in the shared storage.